lib/game/base_game.dart

```diff
@@ -61,32 +60,30 @@ class BaseGame extends Game with FPSCounter {
     if (size != null) {
       c.onGameResize(size);
     }
-
-    c.onMount();
   }
 
-  /// Adds a new component to the components list.
-  ///
-  /// Also calls [preAdd], witch in turn sets the current size on the component (because the resize hook won't be called until a new resize happens).
+  /// Prepares and registers a component to be added on the next game tick
   void add(Component c) {
-    preAdd(c);
-    components.add(c);
+    prepare(c);
+    _addLater.add(c);
   }
 
-  /// Registers a component to be added on the components on the next tick.
-  ///
-  /// Use this to add components in places where a concurrent issue with the update method might happen.
-  /// Also calls [preAdd] for the component added, immediately.
-  void addLater(Component c) {
-    preAdd(c);
-    _addLater.add(c);
+  /// Prepares and registers a list of components to be added on the next game tick
+  void addAll(Iterable<Component> components) {
+    components.forEach(prepare);
+    _addLater.addAll(components);
   }
 
   /// Marks a component to be removed from the components list on the next game loop cycle
```

Comment:
let's unify `on the next game tick` and `on the next game loop cycle` as they both mean the same thing. no preferences on which but we should use consistent language to avoid confusion

Reply:
I took game tick now, but if anyone prefers the other I can change to that.